File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.java
class Solution {
    public int solution(int num) {
        int answer = 0;

        if (num == 1) return 0;

        while (num != 1) {
            if (answer > 400) return -1;
            if (num % 2 == 0) {
                num /= 2;
                answer++;
            } else {
                num *= 3;
                num++;
                answer++;
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.py
def solution(num):
    tried = 0
    result = 0
    while True:
        if tried ==0 and  num ==1:
            result ==0
            break
        tried += 1
        if num % 2 ==0:
            num = num/2
        elif num % 2 == 1:
            num = num*3
            num = num+1

        if num ==1:
            result = tried
            break

        if tried > 500:
            result = -1
            break
    answer = result
    return answer